File: server/models/Order.js
import mongoose, { Types } from 'mongoose';

const OrderSchema = new mongoose.Schema(
  {
    products: [
      {
        productId: { type: String, required: true },
        name: { type: String, rquired: true },
        brand: { type: String, required: true },
        cost: { type: Number, required: true },
        picturePath: { type: String, required: true },
        quantity: { type: Number, required: true },
      },
    ],
    totalAmount: { type: Number, required: true },
    totalQuantity: { type: Number, required: true },
    user: {
      type: Types.ObjectId,
      required: true,
      ref: 'User',
    },
    date: { type: Date, required: true },
  },
  { timestamps: true },
);

const Order = mongoose.model('Order', OrderSchema);

export default Order;

File: mobile_app/lib/models/order.dart
class OrderProducts {
  final String id;
  final String name;
  final String brand;
  final double cost;
  final String picturePath;
  final int quantity;

  OrderProducts({
    required this.id,
    required this.name,
    required this.brand,
    required this.cost,
    required this.picturePath,
    required this.quantity,
  });
}

class Order {
  final String id;
  final String firstName;
  final String lastName;
  final String orderDate;
  final String deliveryDate;
  final int totalQuantity;
  final double totalAmount;
  List<OrderProducts> orderProducts = [];

  Order({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.orderDate,
    required this.deliveryDate,
    required this.totalQuantity,
    required this.totalAmount,
    required this.orderProducts,
  });
}
